### PR TITLE
ft(S3C-3692): Add accountId -> canonicalId conversion for ListRecentMetrics

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

**Review by jonathan-gramain — COMMENTED** [review]
no need for a test?

**Comment by tmacro** [comment]
@jonathan-gramain tests are taking a bit of work, as none currently exist for `ListRecentMetrics`

**Comment by tmacro** [comment]
needs #1070 before tests will run correctly

File: tests/functional/server/testListMetrics.js [hunk 1]
```diff
@@ -18,43 +19,106 @@ describe('Request ranges', function test() {
...
     tests.forEach(test => {
-        const { start, end } = test;
-        it(`should handle a request range of ${end - start}ms`, done => {
-            const params = {
-                timeRange: [start, end],
+        const {
+            timeRange, type, resource, expected,
+        } = test;
+        const msg = timeRange
+            ? `should handle a request range of ${timeRange[0] - timeRange[1]}ms`
```
Comment (saitharunthodupunuru) [inline tests/functional/server/testListMetrics.js]: I think it should be `${timeRange[1] - timeRange[0]}ms`
```suggestion
            ? `should handle a request range of ${timeRange[1] - timeRange[0]}ms`
```

**Review by saitharunthodupunuru — APPROVED** [review]
LGTM except a small msg change

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**